### ISSUE TITLE
Breadcrumb has poor compatibility with assistive technology

### ISSUE THREAD

**Issue by adampage** [adampage/github-cli-test#8]
The HTML+CSS implementation of our Breadcrumb component has poor compatibility with assistive technology and is lacking in semantic quality. To improve it, we can imitate the [WAI-ARIA Breadcrumb design pattern](https://www.w3.org/TR/wai-aria-practices/#breadcrumb) by making the following changes:

## Markup

- Enclose the component in a `<nav>` landmark labeled by `aria-label` or `aria-labelledby`.
- Use an `<ol>` to convey the ordered nature of the link hierarchy.
- Use the `aria-current` attribute to identify the link matching the current page.

## Styles

- Render the `/` separators without CSS pseudo content so that they are not presented to users of assistive technology.

## Applicable WCAG Success Criteria

- [`2.4.4` Link Purpose (In Context) — Level A](https://www.w3.org/WAI/WCAG21/quickref/#link-purpose-in-context)
- [`2.4.8` Location — Level AAA](https://www.w3.org/WAI/WCAG21/quickref/#location)
- [`3.2.3` Consistent Navigation — Level AA](https://www.w3.org/WAI/WCAG21/quickref/#consistent-navigation)
- [`4.1.2` Name, Role, Value — Level A](https://www.w3.org/WAI/WCAG21/quickref/#name-role-value)
